lib/compiler/filters/static-filter-checker.js

```diff
@@ -53,6 +53,14 @@ var checks = {
         }
     },
 
+    isPropertyAccess: {
+        displayName: 'property access',
+
+        test: function(node) {
+            return node.type === 'MemberExpression';
```

Comment:
The check is too benevolent as it allows passing whatever you want in `node.object` and `node.property`. For example, these would be valid filters (assuming `f` is a function):

```juttle
f()[1] < 5
field[f()] < 5
```

We should also check that `node.property` is a simple literal and that `node.object` is either a field or another `MemberExpression` (so that we allow nested member expressions: `a.b.c`, `a[0][1]`, etc.).